packages/back-end/src/routers/segment/segment.controller.ts

```diff
@@ -219,18 +218,17 @@ export const putSegment = async (
     throw new Error("Invalid data source");
   }
 
-  segment.set("datasource", datasource);
-  segment.set("userIdType", userIdType);
-  segment.set("name", name);
-  segment.set("owner", owner);
-  segment.set("sql", sql);
-  segment.set("dateUpdated", new Date());
-
-  await segment.save();
+  await updateSegment(id, org.id, {
+    datasource,
+    userIdType,
+    name,
+    owner,
+    sql,
+    dateUpdated: new Date(),
+  });
 
   res.status(200).json({
     status: 200,
-    segment,
```

Comment:
You might've already tested this but are we sure this isn't used on the frontend in any capacity?

Reply:
From what I can tell, nothing on the frontend was expecting the updated segment to be in the response body, since we're getting the segment data via `useDefinitions`.